testing/smoke-ide-test/src/smokeIdeTest/groovy/org/gradle/ide/sync/AbstractIdeSyncTest.groovy

```diff
@@ -103,22 +108,30 @@ abstract class AbstractIdeSyncTest extends Specification {
         def ideHomeOption = "--ide-home=$ideHome"
         def ideOption = "--ide=$ide"
 
-        def syncProcessBuilder = new ProcessBuilder(findIdeStarter().toString(), gradleDistOption, projectOption, ideHomeOption, ideOption)
-            .directory(testDirectory)
-            .redirectError(ProcessBuilder.Redirect.INHERIT)
-            .redirectOutput(ProcessBuilder.Redirect.INHERIT)
-
-        syncProcessBuilder.environment().put("JAVA_HOME", AvailableJavaHomes.jdk17.javaHome.absolutePath)
-
-        def syncProcess = syncProcessBuilder.start()
-        Runtime.getRuntime().addShutdownHook {
-            try {
-                syncProcess.destroy();
-            } catch (Exception e) {
-                e.printStackTrace();
+        DefaultClientExecHandleBuilder builder = new DefaultClientExecHandleBuilder(
+            TestFiles.pathToFileResolver(), Executors.newCachedThreadPool(), new DefaultBuildCancellationToken()
+        )
+
+        builder
+            .setExecutable(findIdeStarter().toString())
+            .args(gradleDistOption, projectOption, ideHomeOption, ideOption)
+            .setWorkingDir(testDirectory)
+            .setStandardOutput(System.out)
+            .setErrorOutput(System.err)
+            .environment("JAVA_HOME", AvailableJavaHomes.jdk17.javaHome.absolutePath)
+
+        System.err.println("Running IDE sync with: ${builder.commandLine.join(' ')}")
+        def handle = builder.build().start()
+        if (handle.state == ExecHandleState.STARTED) {
+            Runtime.getRuntime().addShutdownHook {
+                if (handle.state == ExecHandleState.STARTED) {
+                    handle.abort()
+                }
             }
         }
-        assert syncProcess.waitFor() == 0
+        def result = handle.waitForFinish()
+        System.err.println("IDE sync process finished: $result")
+        result.rethrowFailure().assertNormalExitValue()
```

Comment:
Reimplementation of external process running by using better tools. Maybe it will reduce flakiness that we were observing for some of `IdeSyncTest`s. 
Kudos to @octylFractal for implementation!